### DISCUSSION TITLE
Has STM32G4 support for USB?

### DISCUSSION THREAD

**Discussion by jgpeiro** [micropython/micropython#11547] [STM32 / Pyboard]
I tried to enable USB on the NUCLEO_G474RE port simply by:
#define MICROPY_HW_ENABLE_USB       (1)

#define MICROPY_HW_USB_FS              (1)
#define MICROPY_HW_USB_VBUS_DETECT_PIN (pin_A9)
#define MICROPY_HW_USB_OTG_ID_PIN      (pin_A10)

and got lot of errors. I saw the code generated with STM32CubeMx and looks to me that this USB uses little bit different USB code than other families so is not compatible with the actual micropython USB stack?

**Comment by jimmo** [comment]
According to the PR that added the G4 nucleo board it's supposed to work, but it definitely doesn't.

See https://github.com/micropython/micropython/pull/11556 which is a quick attempt to make it work (but unfortunately I do not have a G4 board to test it with).

**Comment by jgpeiro** [comment]
Hello

It was more easy than I expected, so most of the code was compatible with already existing code for G0...it makes sense.
I applied your #11556 changes and now the USB works as usual on a custom G473 board.

Thanks
